File: app/src/main/java/info/nightscout/androidaps/plugins/pump/medtronic/defs/BatteryType.java
package info.nightscout.androidaps.plugins.pump.medtronic.defs;

import java.util.HashMap;
import java.util.Map;

import info.nightscout.androidaps.MainApp;
import info.nightscout.androidaps.R;

/**
 * Created by andy on 6/4/18.
 */

public enum BatteryType {

    None(R.string.key_medtronic_pump_battery_no, 0, 0),
    Alkaline(R.string.key_medtronic_pump_battery_alkaline, 1.20d, 1.47d), //
    Lithium(R.string.key_medtronic_pump_battery_lithium, 1.22d, 1.64d), //
    NiZn(R.string.key_medtronic_pump_battery_nizn, 1.40d, 1.70d) //
    ;

    private final String description;
    public double lowVoltage;
    public double highVoltage;

    static Map<String, BatteryType> mapByDescription;

    static {
        mapByDescription = new HashMap<>();

        for (BatteryType value : values()) {
            mapByDescription.put(value.description, value);
        }
    }

    BatteryType(int resId, double lowVoltage, double highVoltage) {
        this.description = MainApp.gs(resId);
        this.lowVoltage = lowVoltage;
        this.highVoltage = highVoltage;
    }

    public static BatteryType getByDescription(String batteryTypeStr) {
        if (mapByDescription.containsKey(batteryTypeStr)) {
            return mapByDescription.get(batteryTypeStr);
        }
        return BatteryType.None;
    }
}

File: pump/medtronic/src/main/kotlin/app/aaps/pump/medtronic/defs/BatteryType.kt
package app.aaps.pump.medtronic.defs

import androidx.annotation.StringRes
import app.aaps.pump.medtronic.R

/**
 * Created by andy on 6/4/18.
 */
@Suppress("SpellCheckingInspection")
enum class BatteryType(val key: String, @StringRes val friendlyName: Int, val lowVoltage: Double, val highVoltage: Double) {

    None("medtronic_pump_battery_no", R.string.medtronic_pump_battery_no, 0.0, 0.0),
    Alkaline("medtronic_pump_battery_alkaline", R.string.medtronic_pump_battery_alkaline, 1.20, 1.47),
    Lithium("medtronic_pump_battery_lithium", R.string.medtronic_pump_battery_lithium, 1.22, 1.64),
    NiZn("medtronic_pump_battery_nizn", R.string.medtronic_pump_battery_nizn, 1.40, 1.70),
    NiMH("medtronic_pump_battery_nimh", R.string.medtronic_pump_battery_nimh, 1.10, 1.40);

    companion object {

        fun getByKey(someKey: String) = BatteryType.entries.firstOrNull { someKey == it.key } ?: None
    }
}